netebpfext/net_ebpf_ext_helpers.h

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file Prototypes and implementations for helper functions that are common to various programs types implemented in
+ * netebpfext.
+ */
+
+#pragma once
+#include "ebpf_platform.h"
+
+// static uint16_t
+// _ntohs(uint16_t us)
+// {
+//     return us << 8 | us >> 8;
+// }
+
+static int
+_partial_sum(_In_ const void* buffer, int buffer_len)
+{
+    int partial_sum = 0;
+    for (int i = 0; i < buffer_len / 2; i++)
+        partial_sum += *((uint16_t*)buffer + i);
+    return partial_sum;
+}
+
+static int
+_net_ebpf_ext_csum_diff(_In_opt_ const void* from, int from_size, _In_opt_ const void* to, int to_size, int seed)
+{
+    int csum_diff = -1;
+
+    if ((from_size % 4 != 0) || (to_size % 4 != 0))
+        // size of buffers should be multiples of 4.
+        goto Exit;
+
+    if ((from == NULL) && (to == NULL)) {
+        // Trivial case.
+        csum_diff = 0;
+        goto Exit;
+    }
+
+    if ((from != NULL && to != NULL) && (from_size == to_size) && RtlCompareMemory(from, to, from_size) == 0) {
```

Comment:
This is ok but it means you're making 2 passes over both buffers data (RtlCompareMemory here and _partial_sum below). It might be faster to just skip this check. Would need to try it both ways to see which is more performant. Not a blocking issue either way.